Week3/More JavaScript.js

```diff
@@ -0,0 +1,180 @@
+
+/**************************** More JavaScript ***************************** */
+
+// 1. Create a function that takes 3 arguments and returns the sum of the these arguments.
+
+function f1(a,b,c) {
+   return a + b + c
+};
+
+console.log(f1(1,2,3)); //  6
+
+// 2. Create a function named colorCar that receives a color, and prints out, 'a red car' for example.
+
+function colorCar(color) {
+    return "a " + color +" car" 
+}
+
+console.log(colorCar('red'));   //  a red car
+
+// 3. Create an object and a function that takes the object as a parameter and prints out all of its properties and values.
+
+const bestDriver = {
+    firstName:"Mika", 
+    lastName:"Hakkinen", 
+    teamName:"McLaren", 
+    age:50};
+    
+function printObj(obj) {
+    for (const key in obj) {
+        console.log(`His ${key} : ${obj[key]}`)    
+    }
+ 
+} 
+
+printObj(bestDriver);   
+
+
+// 4. Create a function named vehicleType that receives a color, and a code, 1 for car, 2 for motorbike. 
+    // And prints 'a blue motorbike' for example when called as vehicleType("blue", 2)
+
+let vehicleCategory = {
+    1 : "car",
+    2 : "motorbike", 
+};
+    function vehicleType(color,category) {
+        console.log("a " + color + " " + vehicleCategory[category] );        
+}
+     
+vehicleType("blue", 2);
+
+// 5. Can you write the following without the if statement, but with just as a single line with console.log(...);?
+
+    /* Yes, we can use ternary operator to achieve it. */
+
+console.log(3 === 3 ? "yes" : "no");
+    
+// 6. Create a function called vehicle, like before, but takes another parameter called age, 
+    //so that vehicle("blue", 1, 5) prints 'a blue used car'
+
+function vehicle(color, category, age) {
+    if (age >= 1) {
+        age = "used";
+    } else {
+        age = "new";
+    }
+    console.log("a " + color + " " + age + " " + vehicleCategory[category]);
+} 
+    
+vehicle("blue", 1, 5);
+
+// 7. Make a list of vehicles, you can add "motorbike", "caravan", "bike", or more.
+
+let listOfVehicle = ["motorbike", "caravan", "truck", "bike", "jeep"];
+
+// 8. How do you get the third element from that list?
+
+console.log(listOfVehicle[2]);
+
+// 9. Change the function vehicle to use the list of question 7. So that vehicle("green", 3, 1) prints "a green new bike".
+
+function newVehicle(color, category, age) {
+    if (age >= 4) {
+        age = "used";
+    } else {
+        age = "new";
+    }
+    console.log("a " + color + " " + age + " " + listOfVehicle[category]);
+}
+
+newVehicle("green", 3, 1);
+
+ 
+// 10. Use the list of vehicles to write an advertisement. So that it prints something like: "Amazing Joe's Garage, we service cars, motorbikes, caravans and bikes.". (Hint: use a for loop.)
+
+listOfVehicle = ["car", "motorbike", "caravan", "bike"];
+
+let nameOfVehicle = "";
+
+for (i = 1; i < listOfVehicle.length; i++) {
+  if (i < listOfVehicle.length - 1) {
```

Comment:
Ah, you almost got it, but now the result will be:

`Amazing Joe's Garage, we service cars, motorbikes, caravans, and bikes.`.

But the `,` before the `and` is not correct. There should not be a `,` before the `and`. So could you change that:)? The rest looks nice!

Reply:
Thanks :) I changed it

Reply:
I fixed the problem. (I fixed also for a similar issue in the next question)